samples/react/src/AppRoot.js

```diff
@@ -22,21 +21,53 @@ export const routePatterns = [
 //    Not needed if not using connected GraphQL.
 // SitecoreContext: provides component resolution and context services via withSitecoreContext
 // Router: provides a basic routing setup that will resolve Sitecore item routes and allow for language URL prefixes.
-const AppRoot = ({ path, Router, graphQLClient }) => {
-  const routeRenderFunction = (props) => <RouteHandler route={props} />;
-  return (
-    <ApolloProvider client={graphQLClient}>
-      <SitecoreContext componentFactory={componentFactory} contextFactory={SitecoreContextFactory}>
-        <Router location={path} context={{}}>
-          <Switch>
-            {routePatterns.map((routePattern) => (
-              <Route key={routePattern} path={routePattern} render={routeRenderFunction} />
-            ))}
-          </Switch>
-        </Router>
-      </SitecoreContext>
-    </ApolloProvider>
-  );
-};
-
-export default AppRoot;
+class AppRoot extends React.Component {
+  state = {
+    ssrRenderComplete: false,
+    contextFactory: new SitecoreContextFactory()
+  }
+
+  setSsrRenderComplete = ssrRenderComplete => {
+    this.setState({
+      ssrRenderComplete
+    })
+  }
+
+  render() {
+    const { path, Router, graphQLClient, ssrState } = this.props;
+
+    if (ssrState && ssrState.sitecore && ssrState.sitecore.route) {
+      // set the initial sitecore context data if we got SSR initial state
+      this.state.contextFactory.setSitecoreContext({
```

Comment:
The `render` method should not have side effects, in any applications using `React.StrictMode` the `render` method will even be run twice in development mode to dissuade you from side effects in `render`.